### PR TITLE
Consistency checker for location service

### PR DESCRIPTION
**Pull request by ibohalsinghkoijam**

### CODE REVIEW

File: src/components/location/location.resolver.ts [hunk 1]
```diff
@@ -124,4 +124,27 @@ export class LocationResolver {
     await this.locationService.delete(id, session);
     return true;
   }
+
```
Comment (candleshine) [inline src/components/location/location.resolver.ts:127]: Make this one resolver

File: src/components/location/location.service.ts [hunk 2]
```diff
@@ -880,4 +887,130 @@ export class LocationService {
       throw new ServerException('Could not delete location');
     }
   }
+
+  async checkZoneConsistency(session: ISession): Promise<boolean> {
```
Comment (candleshine) [inline src/components/location/location.service.ts:891]: Add a 4th function that calls all 3 and does every() on the result

File: test/country.e2e-spec.ts [hunk 3]
```diff
@@ -204,4 +204,27 @@ describe('Country e2e', () => {
 
     expect(locations.items.length).toBeGreaterThanOrEqual(2);
   });
+
```
Comment (candleshine) [inline test/country.e2e-spec.ts:207]: Just call the 1 resolver here

File: test/region.e2e-spec.ts [hunk 4]
```diff
@@ -264,4 +264,30 @@ describe('Region e2e', () => {
 
     expect(locations.items.length).toBeGreaterThanOrEqual(2);
   });
+
+  it('Check consistency across region nodes', async () => {
```
Comment (candleshine) [inline test/region.e2e-spec.ts:268]: remove this

File: test/zone.e2e-spec.ts [hunk 5]
```diff
@@ -184,4 +184,27 @@ describe('Zone e2e', () => {
 
     expect(locations.items.length).toBeGreaterThanOrEqual(2);
   });
+
```
Comment (candleshine) [inline test/zone.e2e-spec.ts:187]: remove this

File: src/components/location/location.resolver.ts (hunk 1 above)
Comment (ibohalsinghkoijam) [reply src/components/location/location.resolver.ts:127]: kept one resolver

File: test/region.e2e-spec.ts (hunk 4 above)
Comment (ibohalsinghkoijam) [reply test/region.e2e-spec.ts]: Removed

File: test/zone.e2e-spec.ts (hunk 5 above)
Comment (ibohalsinghkoijam) [reply test/zone.e2e-spec.ts]: Removed

File: src/components/location/location.service.ts (hunk 2 above)
Comment (ibohalsinghkoijam) [reply src/components/location/location.service.ts:891]: Added 4th function that will handle it

File: test/country.e2e-spec.ts (hunk 3 above)
Comment (ibohalsinghkoijam) [reply test/country.e2e-spec.ts:207]: Done

File: src/components/location/location.service.ts [hunk 6]
```diff
@@ -880,4 +887,150 @@ export class LocationService {
...
+              props: ['name'],
+              nodevar: 'country',
+            });
+          })
+        )
+      ).every((n) => n)
+    );
+  }
+
+  async checkLocationConsistency(
```
Comment (candleshine) [inline src/components/location/location.service.ts:1017]: not a switch. 
every().  call all 3

**Pull request merged**